### ISSUE TITLE
YASGUI server crashes when attempting to query custom SQARQL endpoint

### ISSUE THREAD

**Issue by cicdguy** [YASGUI/Server#2]
Hi Laurens,

I was attempting to use YASGUI.server to enable a proxy to access a CORS-disabled SPARQL endpoint. When I start up the server with the default configurations and run a query while pointing at this endpoint, the server crashes with this message:

```
ubuntu@localhost:~/YASGUI.server-1.1.2$ npm run start


> yasgui-server@1.1.2 start /home/ubuntu/YASGUI.server-1.1.2
> node src/index.js

Running YASGUI on http://localhost:4545
_http_server.js:193
    throw new RangeError(`Invalid status code: ${statusCode}`);
    ^

RangeError: Invalid status code: 0
    at ServerResponse.writeHead (_http_server.js:193:11)
    at ServerResponse._implicitHeader (_http_server.js:158:8)
    at ServerResponse.OutgoingMessage.end (_http_outgoing.js:585:10)
    at ClientRequest.<anonymous> (/home/ubuntu/YASGUI.server-1.1.2/src/corsProxy.js:87:7)
    at emitOne (events.js:77:13)
    at ClientRequest.emit (events.js:169:7)
    at Socket.socketErrorListener (_http_client.js:269:9)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at connectErrorNT (net.js:1016:8)

npm ERR! Linux 4.4.0-75-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "start"
npm ERR! node v4.8.3
npm ERR! npm  v2.15.11
npm ERR! code ELIFECYCLE
npm ERR! yasgui-server@1.1.2 start: `node src/index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the yasgui-server@1.1.2 start script 'node src/index.js'.
npm ERR! This is most likely a problem with the yasgui-server package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node src/index.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs yasgui-server
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls yasgui-server
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/ubuntu/YASGUI.server-1.1.2/npm-debug.log
```
I also get this message on the UI:

```
Unable to get response from endpoint. Possible reasons:

Incorrect endpoint URL
Endpoint is down
Endpoint is not accessible from the YASGUI server and website, and the endpoint is not CORS-enabled
```

I'm using Ubuntu 16.04, npm v2.15.11 and node v4.8.3.

**Comment by LaurensRietveld** [comment]
ah, that's odd. It seems the `0` status code I was sending is not allowed anymore. It expects status codes, i.e. >=100 and <=500. Now returning a 502 (bad gateway) error instead, so it should be resolved. Released this as v1.13

**Comment by LaurensRietveld** [comment]
Forgot to mention issue in commit. For future reference: https://github.com/OpenTriply/YASGUI.server/commit/45730913a42f773c0de7d7ddd8423cc49446bd43
